src/main/java/Jarvis/Parser.java

```diff
@@ -0,0 +1,220 @@
+
+        String nameOfValidCommand = isValidCommand(userInput);
+
+        if (userInput.matches("list")) { // if "list" is entered
+            return Ui.getTaskList(tasks);
+
+        } else if (userInput.matches("bye")) { // if "bye" is entered
+            storage.deleteContents();
+            storage.save(tasks); // save task list to data file after bye is entered
+            return Ui.getByeMessage();
+
+        } else if (userInput.matches("mark \\d+")
+                || userInput.matches("uncheck \\d+")) { // if "mark" or "uncheck" is entered
+
+            int taskNum = Integer.parseInt(userInput.substring(userInput.length() - 1));
+            if (!tasks.isValidTaskNumber(taskNum)) {
+                return null;
+            }
+
+            Task currentTask = tasks.getTask(taskNum - 1);
+            if (userInput.matches("uncheck \\d+")) { // if "uncheck" is entered
+                currentTask.setDone(false);
+                return Ui.getUncheckMessage(currentTask);
+
+            } else { // if "mark" is entered
+                currentTask.setDone(true);
+                return Ui.getMarkMessage(currentTask);
+            }
+
+        } else if (todoMatcher.matches()
+                || deadlineMatcher.matches()
+                || eventMatcher.matches()) { // if task command is entered
+            Task newTask;
+            if (todoMatcher.matches()) { // if "todo" is entered
+                String taskDescription = todoMatcher.group(2);
+                newTask = new ToDo(taskDescription);
+
+            } else if (deadlineMatcher.matches()) { // if "deadline" is entered
+                String taskDescription = deadlineMatcher.group(2);
+                String by = deadlineMatcher.group(3);
+                newTask = new Deadline(taskDescription, by);
+
+            } else { // if "event" is entered
+                String taskDescription = eventMatcher.group(2);
+                String from = eventMatcher.group(3);
+                String to = eventMatcher.group(4);
+                newTask = new Event(taskDescription, from, to);
+
+            }
+            tasks.appendTask(newTask);
+            return Ui.getTaskMessage(tasks, newTask);
+
+        } else if (deleteMatcher.matches()) { // if delete is entered
+            int taskNum = Integer.parseInt(deleteMatcher.group(2));
+            if (tasks.isValidTaskNumber(taskNum)) {
+                Task deletedTask = tasks.getTask(taskNum - 1);
+                tasks.deleteTask(taskNum - 1);
+                return Ui.getDeleteMessage(tasks, deletedTask);
+            }
+
+        } else if (findMatcher.matches()) { // if find is entered
+            String keyword = findMatcher.group(2);
+            return Ui.getFoundTaskMessage(tasks.findTask(keyword));
+
+        } else { // if none of the above commands go through
+            return isWrongFormat(userInput, nameOfValidCommand);
+        }
+        return null;
+    }
+}
```

Comment:
code seems to be a little convoluted, maybe try using enums to replace the if/else statements. For exception handling, i think it would be better if you create separate classes for the different exceptions instead of handling it using if/else